trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/TestPresignedRequestsWithRewrite.java

```diff
@@ -0,0 +1,89 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server;
+
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.google.inject.Inject;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.StringResponseHandler;
+import io.airlift.http.server.testing.TestingHttpServer;
+import io.trino.aws.proxy.server.testing.RequestRewriteUtil;
+import io.trino.aws.proxy.server.testing.TestingS3RequestRewriter;
+import io.trino.aws.proxy.server.testing.TestingUtil.ForTesting;
+import io.trino.aws.proxy.server.testing.containers.S3Container.ForS3Container;
+import io.trino.aws.proxy.server.testing.harness.TrinoAwsProxyTest;
+import io.trino.aws.proxy.server.testing.harness.TrinoAwsProxyTestCommonModules.WithConfiguredBuckets;
+import io.trino.aws.proxy.server.testing.harness.TrinoAwsProxyTestCommonModules.WithTestingHttpClient;
+import io.trino.aws.proxy.spi.credentials.Credentials;
+import io.trino.aws.proxy.spi.rest.S3RequestRewriter;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+import software.amazon.awssdk.services.s3.presigner.model.GetObjectPresignRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PresignedGetObjectRequest;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.Duration;
+
+import static io.airlift.http.client.StringResponseHandler.createStringResponseHandler;
+import static io.trino.aws.proxy.server.testing.RequestRewriteUtil.CREDENTIAL_TO_REDIRECT;
+import static io.trino.aws.proxy.server.testing.RequestRewriteUtil.TEST_CREDENTIAL_REDIRECT_BUCKET;
+import static io.trino.aws.proxy.server.testing.RequestRewriteUtil.TEST_CREDENTIAL_REDIRECT_KEY;
+import static io.trino.aws.proxy.server.testing.TestingUtil.TEST_FILE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TrinoAwsProxyTest(filters = {WithConfiguredBuckets.class, WithTestingHttpClient.class, RequestRewriteUtil.Filter.class})
+public class TestPresignedRequestsWithRewrite
+        extends AbstractTestPresignedRequests
+{
+    @Inject
+    public TestPresignedRequestsWithRewrite(
+            @ForTesting HttpClient httpClient,
+            S3Client internalClient,
+            @ForS3Container S3Client storageClient,
+            @ForTesting Credentials testingCredentials,
+            TestingHttpServer httpServer,
+            TrinoAwsProxyConfig s3ProxyConfig,
+            XmlMapper xmlMapper,
+            S3RequestRewriter requestRewriter)
```

Comment:
If we change `TestingS3RequestRewriter` from an interface to a concrete class this  should be able to request a `TestingS3RequestRewriter` as its bound in the Injector scope and we can avoid the cast